### PR TITLE
Add integration tests for Job Push

### LINKED ISSUE #13516: Add integration tests for Job Push
**Description**
IT tests for the Job Push is missing. To cover the happy path scenario end to end, we need to implement IT tests. ~~These cases should also cover closing a stream which will actually test the `onClose` hook.~~ Please refer to these two discussions for more details: 
https://github.com/camunda/zeebe/pull/13351#discussion_r1263414875
https://github.com/camunda/zeebe/pull/13351#discussion_r1263834385

### PR DESCRIPTION
**Pull request by berkaycanbc**
## Description

IT tests for the Job Push is missing. To cover the happy path scenario end to end, we need to implement IT tests. These cases should also cover closing a stream which will actually test the onClose hook. Please refer to these two discussions for more details:
https://github.com/camunda/zeebe/pull/13351#discussion_r1263414875
https://github.com/camunda/zeebe/pull/13351#discussion_r1263834385

## Related issues

closes #13516 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:96]: :x: This section is flaky for two reasons:

1. We can only push jobs to a stream if the stream existed before the job handled (i.e. created, timed out, failed, etc. - anything that would make it activate-able again).
2. Opening a stream is asynchronous, so there's a chance the process instance is created, and the job is created, before the stream is opened.

There are some options to work around this here, but your best bet is to wait for my PR with actuators. Then you can actually query the broker to ensure the stream exists before creating the process instance.

> **Note**
> In production, since users will be using a job worker, this is not a big deal since we're both streaming and polling.

Comment (npepinpe) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:96]: As this may block you, we can do the other changes for this PR, and then I'm happy to take adapt your PR once mine is merged so you can already do other stuff.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:105]: :x: It's possible for the job batch activated record to be written, but the job to not have been sent or received by the stream consumer yet. Instead, I think it's perfectly fine to just use Awaitility and wait until we receive the expected job within 10 seconds or so. That means we probably don't need the whole ContainerEngine here.

A more complete alternative would be to wait for the job, complete it, and then assert that the process instance is completed. But I don't think it's necessary here.

**Review by npepinpe — CHANGES_REQUESTED** [review]
Thanks for the PR @berkaycanbc!

:x: Unfortunately, it's flaky in two places. I highlighted them in the comments.

:question: I also had one question about the use of containers here. I'm not sure we really need it for this kind of test. It would probably be fine to use the `ClusteringRuleExtension` or something like that.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java
Comment (berkaycanbc) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:49]: I think, using an embedded gateway should just work since we are testing the behaviour here and it does not require that. 👍

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java
Comment (berkaycanbc) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:96]: Alright, these makes sense. Then, let's wait for the blocking PR to be merged. Afterwards, please feel free to add changes.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java
Comment (berkaycanbc) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java:104]: We'll wait #13684 to be merged to properly write assertions to the IT test.

Cc: @npepinpe

**Pull request closed without merging**
